### PR TITLE
Move RegularExpressions contract to be supported from net463.

### PR DESCRIPTION
**Pull request by Priya91**
cc @weshaggard @ericstj @joshfree 

related issue #9061

### CODE REVIEW

**Comment by weshaggard** [comment]
LGTM

Please file a tracking issue to make sure these APIs get added to .NET 4.6.3.

**Comment by joshfree** [comment]
> Please file a tracking issue to make sure these APIs get added to .NET 4.6.3.

+1

**Comment by Priya91** [comment]
I had added netfx-port-consider tag to the original api-review issue #5079, will create a new one as well

**Comment by Priya91** [comment]
OSX machine went offline during CI

**Comment by Priya91** [comment]
@dotnet-bot test Innerloop OSX Debug Build and Test

File: src/System.Text.RegularExpressions/ref/project.json [hunk 1]
```diff
@@ -3,9 +3,9 @@
     "System.Runtime": "4.0.0"
   },
   "frameworks": {
-    "netstandard1.5": {
+    "netstandard1.6": {
       "imports": [
-        "dotnet5.6"
+        "dotnet5.7"
```
Comment (ericstj) [inline src/System.Text.RegularExpressions/ref/project.json]: This doesn't exist (it stopped at 5.6) but that's not material, nuget still falls back to 5.6, then eventually to 5.0, aka dotnet without a version. which is what we used in the old 4.0.0 system.runtime package.

**Comment by ericstj** [comment]
LGTM

**Pull request merged**